From a53b48de4c773f4513584b5fb3bb96074ad0d7e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Sat, 31 Aug 2019 01:57:12 +0200
Subject: [PATCH 009/152] locatePointer: Bind ripples creation to settings

Don't create ripples if locate pointer is not enabled, and bind
creation to the relative desktop interface settings key.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/700
---
 js/ui/locatePointer.js | 20 +++++++++++++++++---
 1 file changed, 17 insertions(+), 3 deletions(-)

diff --git a/js/ui/locatePointer.js b/js/ui/locatePointer.js
index 10caa656d..6ae29419a 100644
--- a/js/ui/locatePointer.js
+++ b/js/ui/locatePointer.js
@@ -11,12 +11,26 @@ const LOCATE_POINTER_SCHEMA = "org.gnome.desktop.interface";
 var LocatePointer = class {
     constructor() {
         this._settings = new Gio.Settings({ schema_id: LOCATE_POINTER_SCHEMA });
-        this._ripples = new Ripples.Ripples(0.5, 0.5, 'ripple-pointer-location');
-        this._ripples.addTo(Main.uiGroup);
+        this._settings.connect(`changed::${LOCATE_POINTER_KEY}`, () => this._syncEnabled());
+        this._syncEnabled();
+    }
+
+    _syncEnabled() {
+        let enabled = this._settings.get_boolean(LOCATE_POINTER_KEY);
+        if (enabled == !!this._ripples)
+            return;
+
+        if (enabled) {
+            this._ripples = new Ripples.Ripples(0.5, 0.5, 'ripple-pointer-location');
+            this._ripples.addTo(Main.uiGroup);
+        } else {
+            this._ripples.destroy();
+            this._ripples = null;
+        }
     }
 
     show() {
-        if (!this._settings.get_boolean(LOCATE_POINTER_KEY))
+        if (!this._ripples)
             return;
 
         let [x, y] = global.get_pointer();
-- 
2.23.0

